backend/Shared/HitMeApp.Shared.Infrastructure/Persistence/DefaultSqlQueryRunner.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Data;
+using System.Threading.Tasks;
+
+namespace HitMeApp.Shared.Infrastructure.Persistence
+{
+    internal sealed class DefaultSqlQueryRunner : ISqlQueryRunner
+    {
+        private readonly ISqlConnectionFactory _connectionFactory;
```

Comment:
We're disposing it multiple times in this class. This is violation of CA2202. Is this common convention when it comes to `IDbConnection`?

Reply:
Technically we're only disposing the newly created connection after a transaction is done. IMHO it happens once in the following manner:

1. `ISqlConnectionFactory` creates a new opened `IDbConnection`
2. `ISqlQueryRunner` runs a passed SQL query
3. `IDbConnection` is disposed  of

As a general rule you should never keep your DB connections open for longer than neccessary to avoid DB locking, memory leaks and all sorts of other unpleasant problems.

That's also the reason for **never** implementing your data access layer in a singleton manner.

---
In case I missed the clue of your comment please elaborate. I might've misunderstood it.

Reply:
Why I meant is that we're disposing it (not closing). Disposing generally means that object will/might be in a state that is inoperative.

I know that `Dispose()` calls `Close()` and nothing bad should happen here (for now), but according to MS guidelines we should never assume `Dispose()` implementation as it might change in the future. 

Maybe we should refactor this to `try`/`finally` with `finally` calling `Close()` and make this whole class implement `IDisposable` to call `IDbConnection` `Dispose()`.

Reply:
Let's keep it like that for a first implementation and ensure connection close in an additional abstraction for transaction management, by maybe using `Unit Of Work` design pattern.